docs/components/sidecar.md

```diff
@@ -0,0 +1,20 @@
+# Sidecar
+
+The sidecar component of Thanos gets deployed along with a Prometheus instance. It implements Thanos' Store API on top of Prometheus' remote-read API and advertises itself as a data source to the cluster. Thereby queriers in the cluster can treat Prometheus servers as yet another source of time series data without directly talking to its APIs.
```

Comment:
worth to mention gossip directly?

Reply:
Mh, I mentioned "within the cluster" in a couple of places as a synonym. Membership in the gossip is literally the definition of the cluster.